Release Java SDK 7.0.0 Blog Post (#246)
serverlessworkflow/serverlessworkflow.github.io: src/content/blog/releases/release-sdk-java-7.0.0.md

---
title: Serverless Workflow Java SDK 7.0.0
author: Francisco Javier Tirado Sarti
date: 2025-01-31
description: >
   Heads up, community! We have released a new Java SDK version compatible with Specification 1.0.0.
---

## Announcing the Release of Serverless Workflow Java SDK 7.0.0

Hello, Serverless Workflow community!

> We're excited to announce the release of the latest Java SDK, version 7.0.0, now compatible with the upcoming Serverless Workflow Specification 1.0.0 and including a reference implementation. The reference implementation provides a ready-to-use runtime that supports the Serverless Workflow Specification. It includes a workflow execution engine, validation utilities, and illustrative examples to help you quickly test and deploy your workflows. 

Reviewer: ```suggestion
We're excited to announce the release of the latest Java SDK, version 7.0.0, which is now compatible with the upcoming Serverless Workflow Specification 1.0.0 and includes a reference implementation. The reference implementation provides a ready-to-use runtime that supports the Serverless Workflow Specification. It includes a workflow execution engine, validation utilities, and illustrative examples to help you quickly test and deploy your workflows. 
```

## Java SDK example

> You can dive right in by using [one of the examples](https://github.com/serverlessworkflow/specification/tree/main/examples) available in the specification repository. Here's a quick example to get you started:

Reviewer: ```suggestion
You can dive right in using [one of the examples](https://github.com/serverlessworkflow/specification/tree/main/examples) available in the specification repository. Here's a quick example to get you started:
```

> To use SDK, simply add it as a dependency to your Maven project:

Reviewer: ```suggestion
To use the SDK, simply add it as a dependency to your Maven project:
```

```xml
<dependency>
    <groupId>io.serverlessworkflow</groupId>
    <artifactId>serverlessworkflow-api</artifactId>
    <version>7.0.0</version>
</dependency>
```

Then, you can write 

```java
// Make sure simple.yaml is in your root folder
try (InputStream in = new FileInputStream("simple.yaml")) {
   Workflow workflow = WorkflowReader.readWorkflow (in, WorkflowFormat.YAML);
    // Once you have the Workflow instance, you can use its API to inspect it
}
```

## Java Reference implementation example 

> You can dive right in by using [one of the examples](https://github.com/serverlessworkflow/sdk-java/tree/main/examples) available in the implementation repository or by checking the [Readme](https://github.com/serverlessworkflow/sdk-java/tree/main/impl). 

Reviewer: ```suggestion
You can dive right in using [one of the examples](https://github.com/serverlessworkflow/sdk-java/tree/main/examples) available in the implementation repository or by checking the [README](https://github.com/serverlessworkflow/sdk-java/tree/main/impl). 
```

Here's a quick example to get you started:

To use reference implementation, simply add it as a dependency to your Maven project:

```xml
<dependency>
      <groupId>io.serverlessworkflow</groupId>
      <artifactId>serverlessworkflow-impl-core</artifactId>
      <version>7.0.0</version>
</dependency>
```

> Then, you can write following code to execute a workflow

Reviewer: ```suggestion
Then, you can write the following code to execute a workflow
```

``` java 
// Make sure simple.yaml is in your src/main/resources folder
try (WorkflowApplication appl = WorkflowApplication.builder().build()) {
      logger.info(
          "Workflow output is {}",
          appl.workflowDefinition(WorkflowReader.readWorkflowFromClasspath("simple.yaml"))
              .instance(Map.of())
              .start()
              .join());
    }
```

## Whats new?

> With this release:

Reviewer: ```suggestion
In this release:
```

> - We update to the latest schema revision. 
> - We improve POJO generation from specification schema by:
>      - Fixing generation of Pojos related with EventComsuption strategy. 
> - We release the first documented version of the [reference implementation](https://github.com/serverlessworkflow/sdk-java/tree/main/impl), completing [phase 1](https://github.com/serverlessworkflow/sdk-java/issues/459).

Reviewer: ```suggestion
- We updated the latest schema revision. 
- We improved POJO generation from specification schema by:
     - Fixing generation of POJOs related to the `EventComsuption` strategy. 
- We released the first documented version of the [reference implementation](https://github.com/serverlessworkflow/sdk-java/tree/main/impl), completing [phase 1](https://github.com/serverlessworkflow/sdk-java/issues/459).
```

### Future plans 

> Check out our plans [here](https://github.com/serverlessworkflow/sdk-java/milestone/2)

Reviewer: ```suggestion
Check out our roadmap [here](https://github.com/serverlessworkflow/sdk-java/milestone/2)
```
